app/javascript/packs/shared/data_grid/DataGrid.jsx

```diff
@@ -0,0 +1,118 @@
+import React, { Component } from "react";
+import PropTypes from "prop-types";
+import ReactDataGrid from 'react-data-grid';
+
+class DataGrid extends Component {
+  constructor(props) {
+    super(props);
+    this.cleanProps = this.cleanProps.bind(this);
+    this.transformColumns = this.transformColumns.bind(this);
+    this.setupDefaultProps = this.setupDefaultProps.bind(this);
+
+    this.transformColumns();
+    this.setupDefaultProps();
+
+    // Store the original rows array, and make a copy that
+    // can be used for modifying eg.filtering, sorting
+    this.state = {
+      originalRows: this.props.data,
+      rows: this.props.data.slice(0)
+    };
+  }
+
+  setupDefaultProps() {
+    // Setup the default props if they're not provided
+    if ('rowGetter' in this.props) {
```

Comment:
The idea I had in mind for those 2 components (`DataGrid`, `DataTable`) was that you can provide some of our own properties that are defined on our end (e.g. `columns` - ideally those props should be as similar as possible for both components, so we can use them interchangably as much as possible), however you can pass-through all the parameters for original components (so, all the properties for `BootstrapTable` and `ReactDataGrid`). And there are many props for those 2 original components.

If I understand React/JS correctly, doing it your way (`let { rowGetter, rowsCount, onGridSort } = this.props`) would throw out all other original component props.

Maybe a wider question for you: do you think the approach I'm defending in this comment is even viable? Or should we do it some other way? I was just trying to make those components semi-reusable - but still customizable by using original component props - and allegedly, [component composition](https://facebook.github.io/react/docs/composition-vs-inheritance.html) is the way to go with React.js.